Refactor of DataProtector Tool Documentation (#38)
iExecBlockchainComputing/documentation-tools: documentation/tools/dataprotector/README.md

# 🔐 DataProtector

The DataProtector tool allows application developers to provide users with unparalleled ownership over their data. End users gain the ability to invoke iExec apps without ever exposing their data to any other party. They gain complete privacy of personally identifiable information (PII) or other sensitive classes of data. 

> This revolutionary approach to data management relies on:

Reviewer: ```suggestion
This approach to data management relies on:
```

* end-to-end encryption of data with access controlled entirely by the owner of the data
* confidential computing technology that ensures only authorized apps are permitted access to a user's data
* smart contracts to manage an iExec application's permissions for a user's encrypted data

The DataProtector SDK encompasses the following suite of methods:

> * **protectData**—safeguards data by encrypting it and recording ownership as an NFT using a smart contract

Reviewer: ```suggestion
* **protectData**—safeguards data by encrypting it and recording ownership as an NFT
```

* **grantAccess**—authorizes an application to process a user's data without exposing the data to any external system or user review

> * **revokeAllAccessObservable**—removes all access granted to any iExec applications for a `protectedData` object containing a user's data

Reviewer: ```suggestion
* **revokeAllAccessObservable**—removes all access granted to any iExec applications or user for a `protectedData` object
```

> * **revokeOneAccess**—removes access granted to one iExec application for a `protectedData` object containing a user's data

Reviewer: ```suggestion
* **revokeOneAccess**—removes a specific access previously granted on a `protectedData`
```
